platform/npm/tests/snfe.js

```diff
@@ -124,6 +124,30 @@ describe('SNFE', () => {
                         Promise.reject({ name: 'easyprivacy', raw: '||example.net/bar/\n^bar.js?' }),
                     ]));
                 });
+
+                it('should not reject on second call in sequence', async () => {
+                    await engine.useLists([
+                        Promise.resolve({ name: 'easylist', raw: '/foo^\n||example.com^' }),
+                        Promise.resolve({ name: 'easyprivacy', raw: '||example.net/bar/\n^bar.js?' }),
+                    ]);
+
+                    await engine.useLists([
+                        Promise.resolve({ name: 'easylist', raw: '/foo^\n||example.com^' }),
+                        Promise.resolve({ name: 'easyprivacy', raw: '||example.net/bar/\n^bar.js?' }),
+                    ]);
+                });
+
+                it('should not reject on second call in parallel', async () => {
+                    engine.useLists([
```

Comment:
@gorhill there's no `await` here. Is this legal? Is this supported? What is the end result?

Reply:
I don't think it should be legal, it's not expected in the current implementation -- `useLists` removes all the filters before adding the new ones.

Reply:
I've committed a change to throw in case this happens.

Reply:
I'm checking for the error here now.

Reply:
For what it's worth, I'm not sure that the promise-based API is worth the trouble.

For example, in an async function one could write:

```js
engine.useLists([
  { name: 'easylist', raw: await fetchEasyList() },
  { name: 'easyprivacy', raw: await fetchEasyPrivacy() },
]);
```

At the moment it's not possible to call `StaticNetFilteringEngine.release()` until the last `StaticNetFilteringEngine#useLists()` is complete.

Reply:
> one could write

This would still require the raw text of filter lists to all sit in memory at once, which is what I want to avoid -- and this is how the extension works. Maybe consider the API low-level and the client code is expected to use it properly -- i.e. calling other methods having an effect on the filterset while `useLists()` is pending results in an undetermined behavior.

Reply:
Sounds good.

If the behavior is undetermined then we should remove the second test here. Let me do this.

Reply:
Done.